tests/test_packages/test_skills/test_weather_ledger.py

```diff
@@ -19,125 +19,125 @@
+
+from tests.conftest import CLI_LOG_OPTION
+
+
+class TestWeatherSkillsFetchaiLedger:
+    """Test that weather skills work."""
+
+    @pytest.fixture(autouse=True)
+    def _start_oef_node(self, network_node):
+        """Start an oef node."""
+
+    @classmethod
+    def setup_class(cls):
+        """Set up the test class."""
+        cls.runner = CliRunner()
+        cls.agent_name_one = "my_weather_station"
+        cls.agent_name_two = "my_weather_client"
+        cls.cwd = os.getcwd()
+        cls.t = tempfile.mkdtemp()
+        os.chdir(cls.t)
+
+    def test_weather(self, pytestconfig):
+        """Run the weather skills sequence."""
+        if pytestconfig.getoption("ci"):
+            pytest.skip("Skipping the test since it doesn't work in CI.")
+        # add packages folder
+        packages_src = os.path.join(self.cwd, 'packages')
+        packages_dst = os.path.join(os.getcwd(), 'packages')
+        shutil.copytree(packages_src, packages_dst)
+
+        # create agent one and agent two
+        result = self.runner.invoke(cli, [*CLI_LOG_OPTION, "create", self.agent_name_one], standalone_mode=False)
+        assert result.exit_code == 0
+        result = self.runner.invoke(cli, [*CLI_LOG_OPTION, "create", self.agent_name_two], standalone_mode=False)
+        assert result.exit_code == 0
+
+        # add packages for agent one and run it
+        agent_one_dir_path = os.path.join(self.t, self.agent_name_one)
+        os.chdir(agent_one_dir_path)
+
+        result = self.runner.invoke(cli, [*CLI_LOG_OPTION, "add", "skill", "weather_station_ledger"], standalone_mode=False)
+        assert result.exit_code == 0
+
+        # change config file of gym connection
+        file_src = os.path.join(self.cwd, 'tests', 'test_packages', 'test_skills', 'data', 'connection.yaml')
+        file_dst = os.path.join(self.t, self.agent_name, 'connections', 'gym', 'connection.yaml')
+        shutil.copyfile(file_src, file_dst)
+
+        process_one = subprocess.Popen([
+            sys.executable,
+            '-m',
+            'aea.cli',
+            "run"
+        ],
+            stdout=subprocess.PIPE,
+            env=os.environ.copy())
+
+        os.chdir(self.t)
+
+        # add packages for agent two and run it
+        agent_two_dir_path = os.path.join(self.t, self.agent_name_two)
+        os.chdir(agent_two_dir_path)
+
+        result = self.runner.invoke(cli, [*CLI_LOG_OPTION, "add", "skill", "weather_client_ledger"], standalone_mode=False)
+        assert result.exit_code == 0
+
+        process_two = subprocess.Popen([
+            sys.executable,
+            '-m',
+            'aea.cli',
+            "run"
+        ],
+            stdout=subprocess.PIPE,
+            env=os.environ.copy())
+
+        # check the gym run ends
+
+        time.sleep(20.0)
+        process_one.send_signal(signal.SIGINT)
+        process_one.wait(timeout=20)
+        process_two.send_signal(signal.SIGINT)
+        process_two.wait(timeout=20)
+
```

Comment:
Ideally, we would check the logs at some point to see if the transaction actually took place.

Reply:
Will work on this.